### PR TITLE
{devel}[fosscuda/2019b] PyTorch v1.6.0 w/ Python 3.7.4

### PR DESCRIPTION
**Pull request by Flamefire**
- [x] Requires https://github.com/easybuilders/easybuild-easyblocks/pull/2104
- [x]  Requires https://github.com/easybuilders/easybuild-easyblocks/pull/2164

### CODE REVIEW

File: easybuild/easyconfigs/p/PyTorch/PyTorch-1.6.0-fosscuda-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    'c3bf8cc8091b24c43aa205a1f2037a518d249eb5997bea7f543b7fc880e80e21',  # onnx-20200531.tar.gz
+    'c7682e9007ddfd94072647abab3e89ffd9084089460ae47d67060974467b58bf',  # benchmark-20180525.tar.gz
+    '314cde420a7cf692bdb6877bc6af6bc514805f6cdb8bee90f32566ed08d94b1c',  # onnx-tensorrt-20190916.tar.gz
+    '2435b7fb83b8a608c24ca677907aa9a35e482a7f018e65ca69481b3c8c9f7caf',  # protobuf-20200214.tar.gz
+    'f6c4dab91ae9a03b3019e7cab0572743afd0e1b6e75b97fcca50259c737c924e',  # psimd-20200517.tar.gz
+    'd13818a10a645d557202aa005606337b2a0cdf690e5655253800ab11abb0c076',  # pthreadpool-20200615.tar.gz
+    '1a1ccf8779332a7d6d0db9034d42df188867cbaf52eb71d74451f79aa8ce2040',  # pybind11-20190204.tar.gz
+    '13100c3deed300bbf16f87d8af3539f432462bfef9d38f0c7e3e387dc2e88676',  # PeachPy-20180218.tar.gz
+    '8cb5fae822077ca9cbc14dcc7bba9a3a35ad519284fc5169f9a176672c63860a',  # sleef-20190730.tar.gz
+    'be111cf161b587812fa3b106fe550efb6f129b8b0b702fef32fac23af9580e5e',  # tbb-20181009.tar.gz
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-1.6.0-fosscuda-2019b-Python-3.7.4.eb:251]: I have an existing download from `https://github.com/intel/tbb/archive` named `tbb-20181009.tar.gz`, which has a different checksum...

The contents of the tarball are exactly the same though (no surprise there), so we could add 2 alternative checksums:

```suggestion
    # checksum for original download of tbb-20181009.tar.gz from https://github.com/intel/tbb/archive
    ('dc0a8d8d96cb8765782aa6ac1b509ad4db955d9bbb58fa5cc2265f0292756d72',
    # checksum for download of tbb-20181009.tar.gz from https://github.com/01org/tbb/archive
     'be111cf161b587812fa3b106fe550efb6f129b8b0b702fef32fac23af9580e5e'),
```

see also #10367

File: easybuild/easyconfigs/p/PyTorch/PyTorch-check-cpp-extension.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-check-cpp-extension.py:1]: @Flamefire Can we add a short description about what this test script does + author or upstream reference?

**Comment by boegel** [comment]
@Flamefire I see one failing test:

```
======================================================================
ERROR: test_set_affinity_in_worker_init (__main__.TestSetAffinity)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_dataloader.py", line 2004, in test_set_affinity_in_worker_init
    for sample in dataloader:
  File "/tmp/eb-qcz48h81/tmposm2a_uv/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 363, in __next__
    data = self._next_data()
  File "/tmp/eb-qcz48h81/tmposm2a_uv/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 989, in _next_data
    return self._process_data(data)
  File "/tmp/eb-qcz48h81/tmposm2a_uv/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 1014, in _process_data
    data.reraise()
  File "/tmp/eb-qcz48h81/tmposm2a_uv/lib/python3.7/site-packages/torch/_utils.py", line 395, in reraise
    raise self.exc_type(msg)
OSError: Caught OSError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/tmp/eb-qcz48h81/tmposm2a_uv/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 142, in _worker_loop
    init_fn(worker_id)
  File "test_dataloader.py", line 1992, in worker_set_affinity
    os.sched_setaffinity(0, [2])
OSError: [Errno 22] Invalid argument
```

**Comment by Flamefire** [comment]
No idea. `os.sched_setaffinity(0, [2])` looks like a valid call. Or wait: Do you happen to have only 1 CPU (or less than 3) allocated? Nice PyTorch! Really nice!

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 5 out of 6 (1 easyconfigs in this PR)
taurusml4 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/ddacc707b0a8cc112ae024b063acea73 for a full test report.

**Comment by boegel** [comment]
> No idea. `os.sched_setaffinity(0, [2])` looks like a valid call. Or wait: Do you happen to have only 1 CPU (or less than 3) allocated? Nice PyTorch! Really nice!
> 
> Guess we should disable this test then?

I was tested in a Slurm job, so definitely with restricted access to cores, but I certainly had more than 1 core...
Or do you mean sockets?

**Comment by Flamefire** [comment]
According to https://docs.python.org/3/library/os.html#os.sched_setaffinity it is CPUs. So no idea... Maybe because you don't have access to CPU 2?

**Comment by boegel** [comment]
Tried again in a Slurm job with 8 cores (out of 36 available, in a 2-socker 18-core proc system), now `test_set_affinity_in_worker_init` passed. But another test failed:

```
======================================================================
FAIL: test_lstm (quantization.test_backward_compatibility.TestSerialization)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/eb-1hbt9lal/tmpiuhh9sss/lib/python3.7/site-packages/torch/testing/_internal/common_quantized.py", line 124, in test_fn
    qfunction(*args, **kwargs)
  File "/tmp/vsc40023/easybuild_build/PyTorch/1.6.0/fosscuda-2019b-Python-3.7.4/pytorch-1.6.0/test/quantization/test_backward_compatibility.py", line 132, in test_lstm
    self._test_op(mod, input_size=[4, 4, 3], input_quantized=False, generate=False, new_zipfile_serialization=True)
  File "/tmp/vsc40023/easybuild_build/PyTorch/1.6.0/fosscuda-2019b-Python-3.7.4/pytorch-1.6.0/test/quantization/test_backward_compatibility.py", line 68, in _test_op
    self.assertEqual(qmodule(input_tensor), expected, atol=prec)
  File "/tmp/eb-1hbt9lal/tmpiuhh9sss/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 1111, in assertEqual
    exact_dtype=exact_dtype, exact_device=exact_device)
  File "/tmp/eb-1hbt9lal/tmpiuhh9sss/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 1085, in assertEqual
    self.assertTrue(result, msg=msg)
AssertionError: False is not true : Tensors failed to compare as equal! With rtol=1.3e-06 and atol=1e-05, found 13 element(s) (out of 112) whose difference(s) exceeded the margin of error (including 0 nan comparisons). The greatest difference was 0.9640435565029293 (4.41188467448228e-06 vs. 0.9640479683876038), which occurred at index (3, 0, 6).

----------------------------------------------------------------------
Ran 276 tests in 415.229s

FAILED (failures=1, skipped=7)
```

**Comment by Flamefire** [comment]
-.- No idea. Another test to exclude I think. Reported it to pytorch: https://github.com/pytorch/pytorch/issues/43209

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
taurusml9 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/772bc850c5f79aa9aec0c46c403a3347 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
taurusi6208.taurus.hrsk.tu-dresden.de - Linux RHEL 7.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e66e88b79f28142aaab608209555ff94 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
taurusa5 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz, Python 2.7.5
See https://gist.github.com/57506bf64042198748ff916265e2ec54 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
ninhursaga.uio.no - Linux RHEL 8.2, x86_64, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/34bfbc21291ef5224ac11406afc1ad84 for a full test report.

**Comment by Flamefire** [comment]
@terjekv Same issue as https://github.com/easybuilders/easybuild-easyconfigs/pull/11041#issuecomment-675495527

Not sure what went wrong there and hard to investigate as I'm not seeing it on any of our systems... Looks bad to me but don't know what to do about it except disabling the whole test suite this test is in

**Comment by Flamefire** [comment]
@terjekv Added that although I don't feel comfortable doing so. So at least added a dangerous sounding comment...

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
taurusml8 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/aa99787c8e4b48d9576e0daeff849345 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
taurusml8 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/c076621bb58089fda9e6fdf02e2e7a75 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
taurusml8 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/7f87f8cec657943cf2fb8aab7a840a93 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
taurusml8 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/cab0461dfaee9afd4bcff3d23a23192a for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
taurusml24 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/b2cefcd7058e8bc13b389e3931484abf for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
taurusa7 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz, Python 2.7.5
See https://gist.github.com/f875edcd2065d4dd0e8dde317d52b378 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node3415.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/a27f46dda95db5c8a6ebef4b5c146093 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3404.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/0cd5a2e02a4990aaf96d6e9c4227520c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3300.joltik.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/82356d0a556c41096b59816707abfa2a for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @Flamefire!

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
taurusml23 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/f937a2145648cbf296d6e9965b5c628a for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 5 out of 6 (1 easyconfigs in this PR)
taurusml20 - Linux RHEL 7.6, POWER, 8335-GTX, Python 2.7.5
See https://gist.github.com/dbfa9bb9b123f76498f3099d8d58af14 for a full test report.

**Pull request merged**